django/core/mixins.py

```diff
@@ -268,6 +270,32 @@ def _validate_content_object(self, instance):
                     f"instance {instance} does not have a {field} attribute"
                 )
 
+    @action(detail=True, methods=["post"])
```

Comment:
the action decorator can take in `permission_classes` to override the viewset level permissions https://www.django-rest-framework.org/api-guide/viewsets/